### PR TITLE
Updated Divider as UIView

### PR DESCRIPTION
**Pull request by brennobemoura**
Updated Divider to be a UIView class using layout constraints. This approach turns divider more flexible to constraints updates from superview.

### CODE REVIEW

**Comment by danieldahan** [comment]
On initial glance, this looks great. @adamdahan will comment further, and thank you @brennobemoura for sharing this with us.

**Comment by adamdahan** [comment]
@brennobemoura in review.

**Comment by adamdahan** [comment]
@brennobemoura Do you have any sample code for how you animate the divider which cause the original problem mentioned in ticket: [#1252](https://github.com/CosmicMind/Material/issues/1252)?

**Comment by brennobemoura** [comment]
I will search here. The problem happened around March or April and since them I write this version for Divider and it works. I remember to use the Skeleton view and when it's stops animating the divider was in wrong place or in a different size from view.

**Comment by brennobemoura** [comment]
Today I will search in my code or try to reproduce the error, just wait a little

**Comment by brennobemoura** [comment]
Divider overrode:
Divider overrode

Divider from Material:
Divider from Material

File: Sources/iOS/Divider/Divider.swift [hunk 1]
```diff
@@ -34,183 +34,179 @@ public enum DividerAlignment: Int {
...
+        self.removeFromSuperview()
+        superview.addSubview(self)
+        self.layer.zPosition = 5000
+        
+        self.thickness = thickness
+        
+        switch self.dividerAlignment {
+        case .bottom, .top:
+            superview.layout(self)
+                .leading(c.left, { _, _ in .equal })
```
Comment (OrkhanAlikhanov) [inline Sources/iOS/Divider/Divider.swift]: See https://github.com/CosmicMind/Material/pull/1236 to learn about layout relations. You can reduce this to  `.leading(c.left)`

File: Sources/iOS/Divider/Divider.swift [hunk 2]
```diff
@@ -34,183 +34,179 @@ public enum DividerAlignment: Int {
...
+                .trailing(c.right, {_, _ in .equal })
```
Comment (OrkhanAlikhanov) [inline Sources/iOS/Divider/Divider.swift]: See #1236 to learn about layout relations. You can reduce this to .leading(c.left)

File: Sources/iOS/Type/EdgeInsets.swift [hunk 3]
```diff
@@ -93,98 +93,116 @@ public func EdgeInsetsPresetToValue(preset: EdgeInsetsPreset) -> EdgeInsets {
...
+        self.init(top: vertical, left: 0, bottom: vertical, right: 0)
+    }
+
+    init(horizontal: CGFloat) {
+        self.init(top: 0, left: horizontal, bottom: 0, right: horizontal)
+    }
+    
+    init(square: CGFloat) {
+        self.init(top: square, left: square, bottom: square, right: square)
+    }
```
Comment (OrkhanAlikhanov) [inline Sources/iOS/Type/EdgeInsets.swift]: I like these additions but would be better if you had done these changes in separate commits so that we could see changes separately with a nice commit message instead of sorting the changes in our mind.

File: Sources/iOS/Divider/Divider.swift [hunk 4]
```diff
@@ -34,183 +34,179 @@ public enum DividerAlignment: Int {
...
     }
-  }
 }
 
-/// A memory reference to the Divider instance.
-fileprivate var DividerKey: UInt8 = 0
-
-extension UIView {
-  /// TabBarItem reference.
-  public private(set) var divider: Divider {
```
Comment (OrkhanAlikhanov) [inline Sources/iOS/Divider/Divider.swift]: It seems now instead of `view.divider` we need to do `Divider.from(view: view)`. This is a radical API  change

Comment (danieldahan) [reply Sources/iOS/Divider/Divider.swift]: I personally do not like the `Divider.from(view: view)` API change at all. Maybe we need to reconsider the entire `divider` on a `UIView` completely. @OrkhanAlikhanov @adamdahan

Comment (adamdahan) [reply Sources/iOS/Divider/Divider.swift]: Yup I agree with both of you @OrkhanAlikhanov @danieldahan I like some of the refactoring done but overall the impact of this change is to large for existing users of Material.

File: Sources/iOS/Type/EdgeInsets.swift (hunk 3 above)
Comment (adamdahan) [reply Sources/iOS/Type/EdgeInsets.swift]: This is a nice little clean up.

**Comment by adamdahan** [comment]
@danieldahan @OrkhanAlikhanov overall I think we should close this PR for now as it is a "one off situation" for @brennobemoura issue with his app. Let's utilize the idea's @brennobemoura presented for clean up and constraint based layout and figure out our overall position with Divider for our next big updates.

**Comment by danieldahan** [comment]
@adamdahan I agree. This PR has presented some really nice ideas to consider, and I like that a lot, so thank you @brennobemoura.
